### PR TITLE
WIP: releng improvements

### PR DESCRIPTION
**Pull request by barthalion**
This pull request switches the project to poetry for dependencies management, which should get us reasonable local development workflow and easier dependabot upgrades. Also, updates everything to the latest versions, so I don't expect test suite to still work.

Additionally, execute smoke tests in an environment created by docker-compose – this is both to make it easier to run these locally and ensure it behaves the same on CI and locally.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -2,26 +2,33 @@ FROM python:3.9 as builder
...
+
+RUN flatpak --user remote-add flathub https://flathub.org/repo/flathub.flatpakrepo && \
+    flatpak --user remote-add flathub-beta https://flathub.org/beta-repo/flathub-beta.flatpakrepo
 
 COPY ./app /app
 COPY ./data /data
 COPY --from=builder /venv /venv
 
-ENTRYPOINT ["/venv/bin/uvicorn", "--app-dir", "/", "app.main:app", "--host", "0.0.0.0", "--port", "8000", "--reload", "--reload-dir", "/app", "--reload-dir", "/data"]
+ENTRYPOINT ["uvicorn", "app.main:app", "--host", "0.0.0.0", "--port", "8000"]
```
Comment (razzeee) [inline Dockerfile]: I don't think we want the reloading to go missing or is this image no longer used for dev?

File: tests/main.py [hunk 2]
```diff
@@ -185,12 +190,16 @@ def test_feed_by_new():
     feed = etree.fromstring(response.text.encode("utf-8"))
     expected = _get_expected_xml_result("test_feed_by_new")
 
-    # Remove runtime-generated dates
+    # Remove runtime-generated date and title because encoding is hard
```
Comment (razzeee) [inline tests/main.py:193]: I don't think this will be understandable/useful. What's actually happening here/the problem?

File: Dockerfile (hunk 1 above)
Comment (razzeee) [reply Dockerfile]: Well, seeing it in context, it might be problematic with this container also being used for smoketests now.

**Comment by razzeee** [comment]
Added some nitpicks and it feels like running smoketests now needs docs, as it's not easy to figure out (?)

Not thrilled, that I can't use the container to do actual dev work anymore (no need to pollute the local env with deps)

File: Dockerfile (hunk 1 above)
Comment (barthalion) [reply Dockerfile]: `--reload` is still passed to uvicorn in docker-compose: https://github.com/flathub/backend/blob/master/docker-compose.yml#L13, so when you do `docker compose up`:

```
backend_1  | INFO:     Will watch for changes in these directories: ['/app']
backend_1  | INFO:     Uvicorn running on http://0.0.0.0:8000 (Press CTRL+C to quit)
backend_1  | INFO:     Started reloader process [1] using watchgod
backend_1  | INFO:     Started server process [8]
backend_1  | INFO:     Waiting for application startup.
backend_1  | INFO:     Application startup complete.
```

This shouldn't also cause problems with CPU load as it used `watchgod` now.

File: tests/main.py (hunk 2 above)
Comment (barthalion) [reply tests/main.py:193]: Every year or so something within pytest/feedgen/whatever combo changes and either escapes the long dash `–` or not. I don't wish to spend time on that as it's not the point of that test anyway.

**Comment by barthalion** [comment]
> it feels like running smoketests now needs docs,

Yep, I will add more documentation today. In the meantime, it's described here: https://github.com/flathub/backend/blob/master/.github/workflows/ci.yml#L62

> Not thrilled, that I can't use the container to do actual dev work anymore

Not sure what you mean here? It is still possible with docker compose – one thing I can change is to switch the `builder` stage to include all dev dependencies and make docker-compose use it. I guess we should talk about how your workflow looks like.

File: Dockerfile (hunk 1 above)
Comment (razzeee) [reply Dockerfile]: Somehow missed that change. Should be fine then.

File: tests/main.py (hunk 2 above)
Comment (razzeee) [reply tests/main.py:193]: Well, that's probably enough documentation, for anybody that actually want's to find it. ;)

**Comment by razzeee** [comment]
Nevermind, I think you addressed all. Concerns :)

**Pull request merged**